### PR TITLE
Update: Convert to elide-spring-boot (5.x)

### PR DESCRIPTION
**Pull request by jkusa**
## Description

Convert to elide-spring-boot (5.x)

## Proposed Changes

- Migrate to elide-spring-boot
- Upgrade models to elide 5.x contracts
- Upgrade to Gradle 5.6
- Upgrade to JUnit 5
- Remove getter annotations
- Build updates
- Use elide-spring-boot version of Swagger/GraphiQL (most likely to revisit maybe webjars) 

**Note**: AuthN & AuthZ will need to be refactored/enhanced in another PR.

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/webservice/app/src/main/resources/application.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+elide:
+  modelPackage: 'com.yahoo.navi.ws.models.beans'
```
Comment (aklish) [inline packages/webservice/app/src/main/resources/application.yaml:2]: I don't think this is necessary.

File: packages/webservice/app/src/main/resources/application.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    path: /api/v1
+    enabled: true
+  graphql:
+    path: /graphql/api/v1
+    enabled: false
+  swagger:
+    path: /doc
+    enabled: true
+  async:
+    enabled: false
```
Comment (aklish) [inline packages/webservice/app/src/main/resources/application.yaml:13]: Don't you want async turned on?

File: packages/webservice/app/src/main/resources/application.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+spring:
+  application:
+    name: NaviWS
+  jpa:
+    hibernate:
+      show_sql: true
+      naming:
+        physical-strategy: 'org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl'
+      dialect: 'org.hibernate.dialect.H2Dialect'
```
Comment (aklish) [inline packages/webservice/app/src/main/resources/application.yaml:23]: So you can add "profiles" to the application.yaml for different environments (like prod, test, etc). Is there any value in having something like that for the core repo?

File: packages/webservice/app/src/main/resources/logback-access-spring.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    <appender name="ACCESSFILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>${logDir}/access.log</file>
+        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
+            <fileNamePattern>${logDir}/archived/access_%d{yyyy-MM-dd}.log</fileNamePattern>
+            <maxHistory>30</maxHistory>
+            <totalSizeCap>100MB</totalSizeCap>
+        </rollingPolicy>
+        <encoder>
+            <pattern>%t{yyyy-MM-dd:HH:mm:ss Z} %remoteIP %user %requestURL %statusCode %bytesSent %elapsedTime %header{X-B3-TraceId} %requestContent %responseContent</pattern>
```
Comment (aklish) [inline packages/webservice/app/src/main/resources/logback-access-spring.xml:13]: This logs the response bodies - is that OK for security?

File: packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/DashboardWidget.kt [hunk 5]
```diff
@@ -25,21 +27,18 @@ import javax.persistence.GenerationType
...
-@UpdatePermission(expression = "is an author now OR is an editor now")
-@DeletePermission(expression = "is an author now OR is an editor now")
+@CreatePermission(expression = "$IS_AUTHOR OR $IS_EDITOR")
+@UpdatePermission(expression = "$IS_AUTHOR OR $IS_EDITOR")
+@DeletePermission(expression = "$IS_AUTHOR OR $IS_EDITOR")
 class DashboardWidget : HasAuthor, HasEditors {
-    @get:Id
-    @get:GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (aklish) [inline packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/DashboardWidget.kt:41]: A while back we had a discussion about the best way to do hashcode/equals for Elide models.  We should probably follow that here.

File: packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/User.kt [hunk 6]
```diff
@@ -4,85 +4,84 @@
...
+        joinColumns = [JoinColumn(name = "user_id", referencedColumnName = "id")],
+        inverseJoinColumns = [JoinColumn(name = "dashboard_id", referencedColumnName = "id")]
     )
-    var favoriteDashboards: Collection<Dashboard> = arrayListOf()
+    var favoriteDashboards: MutableSet<Dashboard> = mutableSetOf()
 
-    @get:OneToMany(targetEntity = Role::class)
-    override var roles: Collection<Role> = arrayListOf()
+    @OneToMany
+    override var roles: MutableSet<Role> = mutableSetOf()
```
Comment (aklish) [inline packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/User.kt:86]: Who can edit what roles a user belongs to?

File: packages/webservice/app/src/main/resources/application.yaml (hunk 2 above)
Comment (jkusa) [reply packages/webservice/app/src/main/resources/application.yaml:13]: I was trying to find a why to get rid of the analytic models in swagger, this didn't change anything. I can clean this up.

File: packages/webservice/app/src/main/resources/application.yaml (hunk 3 above)
Comment (jkusa) [reply packages/webservice/app/src/main/resources/application.yaml:23]: Yeah, I was planning to add template profiles. I don't want to make this PR too big. Made an issue for this: https://github.com/yahoo/navi/issues/864

File: packages/webservice/app/src/main/resources/logback-access-spring.xml (hunk 4 above)
Comment (jkusa) [reply packages/webservice/app/src/main/resources/logback-access-spring.xml:13]: Most likely users will need to configure this to the way they want. I pulled this from the spring-elide example. Do you think this is a bad default? Could be helpful for debugging when starting out.

File: packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/DashboardWidget.kt (hunk 5 above)
Comment (jkusa) [reply packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/DashboardWidget.kt:41]: Yes, that did come to mind when refactoring, especially when using `Set`s.

This technique: https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/

Created an issue: https://github.com/yahoo/navi/issues/865

File: packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/User.kt (hunk 6 above)
Comment (jkusa) [reply packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/User.kt:86]: At the moment, nobody. We need to add a prefab `admin` role and allow that role to make user updates (along with other things)

File: packages/webservice/app/src/main/resources/logback-access-spring.xml (hunk 4 above)
Comment (aklish) [reply packages/webservice/app/src/main/resources/logback-access-spring.xml:13]: I don't think we want to put presto data in our logs.

Comment (jkusa) [reply packages/webservice/app/src/main/resources/logback-access-spring.xml:13]: This is the example repo

**Pull request merged**